test/OpenFeature.E2ETests/Steps/EvaluationStepDefinitions.cs

```diff
@@ -41,15 +42,14 @@ public class EvaluationStepDefinitions
         public EvaluationStepDefinitions(ScenarioContext scenarioContext)
         {
             _scenarioContext = scenarioContext;
-            var flagdProvider = new FlagdProvider();
-            Api.Instance.SetProviderAsync(flagdProvider).Wait();
-            client = Api.Instance.GetClient();
         }
 
-        [Given(@"a provider is registered with cache disabled")]
-        public void Givenaproviderisregisteredwithcachedisabled()
+        [Given(@"a provider is registered")]
+        public void GivenAProviderIsRegistered()
         {
-
+            var memProvider = new InMemoryProvider(e2eFlagConfig);
+            Api.Instance.SetProviderAsync(memProvider).Wait();
+            client = Api.Instance.GetClient();
```

Comment:
All the tests now use the in-memory provider (configuration is at the end of this file).